java-type-checker/java_type_checker/expressions.py

```diff
@@ -88,7 +121,51 @@ def __init__(self, receiver, method_name, *args):
         self.method_name = method_name
         self.args = args
 
+    def static_type(self):
+        receiver_type = self.receiver.static_type()
+        method = receiver_type.method_named(self.method_name)
+        return method.return_type
+    
+    def check_types(self):
 
+        self.receiver.check_types()
+        for arg in self.args:
+            arg.check_types()
```

Comment:
Your version of checking arg types is much more concise because it doesn't repeat code, but mine still works even though it is less readable